examples/Planning/ForecastReach.php

```diff
@@ -0,0 +1,389 @@
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        self::showPlannableLocations($googleAdsClient);
+
+        self::showPlannableProducts($googleAdsClient, self::LOCATION_ID);
+
+        self::forecastManualMix(
+            $googleAdsClient,
+            $customerId,
+            self::LOCATION_ID,
+            self::CURRENCY_CODE,
+            self::BUDGET_MICROS
+        );
+
+        self::forecastSuggestedMix(
+            $googleAdsClient,
+            $customerId,
+            self::LOCATION_ID,
+            self::CURRENCY_CODE,
+            self::BUDGET_MICROS
+        );
+    }
+
+    /**
+     * Shows map of plannable locations to their IDs.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     */
+    private static function showPlannableLocations(GoogleAdsClient $googleAdsClient)
+    {
+        $response = $googleAdsClient->getReachPlanServiceClient()->listPlannableLocations();
+
+        printf("Plannable Locations:%sName,\tId,\tParentCountryId%s", PHP_EOL, PHP_EOL);
+        foreach ($response->getPlannableLocations() as $location) {
+            /** @var PlannableLocation $location */
+            printf(
+                "\"%s\",\t%s,\t%s%s",
+                $location->getNameUnwrapped(),
```

Comment:
You can use single quote inside, if you need to use double quote outside.
That's easier to read.

Reply:
Good point, ty.